pandas/core/internals.py

```diff
@@ -978,7 +984,7 @@ def f(m, v, i):
 
         return [self.make_block(new_values, fastpath=True)]
 
-    def coerce_to_target_dtype(self, other):
+    def coerce_to_target_dtype(self, other, copy=False):
```

Comment:
hmm, I dont think this actually helps, why did you add it?

Reply:
Sparse `astype(dtype, copy=False)` makes reinterpret cast, so I overrides `coerce_to_target_dtype` and set `copy=True` in `SparseBlock` class.
https://github.com/pandas-dev/pandas/pull/17386/files/2d94fbaccb5c2bd10a90b2ed77dcb3bade2b21c0#diff-e705e723b2d6e7c0e2a0443f80916abfR2690

Reply:
Sparse `astype(dtype, copy=False)` makes reinterpret cast, so I override `coerce_to_target_dtype` and set `copy=True` in `SparseBlock` class.
https://github.com/pandas-dev/pandas/pull/17386/files/2d94fbaccb5c2bd10a90b2ed77dcb3bade2b21c0#diff-e705e723b2d6e7c0e2a0443f80916abfR2690